core/uemis-downloader.c

```diff
@@ -76,6 +76,68 @@ static int dive_to_read = 0;
 
 static int max_deleted_seen = -1;
 
+/* Linked list to remember already executed divespot download requests */
+struct divespot_mapping {
+	int divespot_id;
+	uint32_t dive_site_uuid;
+	struct divespot_mapping *next;
+};
+static struct divespot_mapping *divespot_mapping = NULL;
+
+static void erase_divespot_mapping()
+{
+	struct divespot_mapping *tmp;
+	while (divespot_mapping != NULL) {
+		tmp = divespot_mapping;
+		divespot_mapping = tmp->next;
+		free(tmp);
+	}
+	divespot_mapping = NULL;
+}
+
+static void add_to_divespot_mapping(int divespot_id, uint32_t dive_site_uuid)
+{
+	struct divespot_mapping *ndm = (struct divespot_mapping*)calloc(1, sizeof(struct divespot_mapping));
+	struct divespot_mapping **pdm = &divespot_mapping;
+	struct divespot_mapping *cdm = *pdm;
+	
+	while (cdm && cdm->next) {
+		cdm = cdm->next;
+	}
+		
+	ndm->divespot_id = divespot_id;
+	ndm->dive_site_uuid = dive_site_uuid;
+	ndm->next = NULL;
+	if (cdm)
+		cdm->next = ndm;
+	else
+		cdm = *pdm = ndm;
+}
+
+static bool is_divespot_mappable(int divespot_id)
+{
+	struct divespot_mapping *dm = divespot_mapping;
+	while (dm) {
+		if (dm->divespot_id == divespot_id) {
+			return true;
+		}
```

Comment:
doesn't need { }

Reply:
You want to keep (or improve) the code quality and I totally understand that. I just need some practice to get in the work flow. I did all the changes you have requested. Thank you for your patience!